tests/recovery.test.ts

```diff
@@ -53,8 +65,46 @@ describe("recovery-tests", () => {
       await LocalRecovery.promise1;
       return ctxt.authenticatedUser;
     }
+
+    static recoveryCount = 0;
+    static readonly maxRecoveryAttempts = 5;
+    static deadLetterResolve: () => void;
+    static deadLetterPromise = new Promise<void>((resolve) => {
+      LocalRecovery.deadLetterResolve = resolve;
+    });
+
+
+    @Workflow({maxRecoveryAttempts: LocalRecovery.maxRecoveryAttempts})
+    static async deadLetterWorkflow(_ctxt: WorkflowContext) {
+      LocalRecovery.recoveryCount += 1
+      await LocalRecovery.deadLetterPromise;
+    }
   }
 
+  test("dead-letter-queue", async () => {
+    LocalRecovery.cnt = 0;
+    const dbosExec = (testRuntime as TestingRuntimeImpl).getDBOSExec();
+
+    const handle = await testRuntime.startWorkflow(LocalRecovery).deadLetterWorkflow();
+
+    for (let i = 0; i < LocalRecovery.maxRecoveryAttempts * 2; i++) {
+      await dbosExec.recoverPendingWorkflows();
+      expect(LocalRecovery.recoveryCount).toBeGreaterThanOrEqual(Math.min(i, LocalRecovery.maxRecoveryAttempts));
+      expect(LocalRecovery.recoveryCount).toBeLessThanOrEqual(LocalRecovery.maxRecoveryAttempts);
+    }
+
+    for (let i = 0; i < LocalRecovery.maxRecoveryAttempts * 2; i++) {
+      await testRuntime.startWorkflow(LocalRecovery, handle.getWorkflowUUID()).deadLetterWorkflow();
+      expect(LocalRecovery.recoveryCount).toBe(i + LocalRecovery.maxRecoveryAttempts + 1);
+    }
+
+    const { rows } = await systemDBClient.query<{status: string, workflow_retries: number}>(`SELECT status, workflow_retries FROM dbos.workflow_status WHERE workflow_uuid=$1`, [handle.getWorkflowUUID()]);
+    expect(rows[0].workflow_retries).toBe(String(LocalRecovery.maxRecoveryAttempts));
+    expect(rows[0].status).toBe(StatusString.DEADLETTER);
+
+    LocalRecovery.deadLetterResolve();
```

Comment:
So looks like even if the workflow is in the DLQ it's not cancelled (still pending) -- in a future PR we might add a way to stop/cancel a workflow.

Also, check the workflow's final status here? I assume it will be `SUCCESS` not DLQ.

Reply:
Yeah, that's by design, we'll add the final check.

Reply:
+1 for checking the status